zyte_common_items/items.py

```diff
@@ -712,3 +713,38 @@ class RealEstate(Item):
 
     #: Contains metadata about the data extraction process.
     metadata: Optional[Metadata] = None
+
+
+@attrs.define(kw_only=True)
+class ProductNavigation(Item):
+    """Represents navigations for a page of product listings on a e-commerce website"""
+
+    #: Main URL from which the data is extracted.
+    url: str = attrs.field(converter=url_to_str)
+
+    #: Name of the category/page with the product list.
+    #:
+    #: Format:
+    #:
+    #: - trimmed (no whitespace at the beginning or the end of the description string),
+    categoryName: Optional[str] = None
+
+    #: List of sub-category links ordered by their position in the page.
+    subCategories: Optional[List[Navigation]] = None
+
+    #: List of product links found on the page category ordered by their position in the page.
+    items: Optional[List[Navigation]] = None
+
+    #: The navigation to the next page, if available.
+    nextPage: Optional[Navigation] = None
+
+    #: Number of the current page.
+    #:
+    #: It should only be extracted if the webpage shows a page number.
+    #:
+    #: It must be 1-based. For example, if the first page of a listing is
+    #: numbered as 0 on the website, it should be extracted as `1` nonetheless.
+    pageNumber: Optional[int] = None
+
+    #: Data extraction process metadata.
+    metadata: Optional[Metadata] = None
```

Comment:
I think we need to clarify the definition of metadata fields. E.g. in the schema @lopuhin published there is no probability in the metadata, so it's not the same data structure as e.g. for Product. The metadata type for subCategories / items might be different as well. 

TBH, I'd prefer to use less distinct Metadata classes.

Reply:
My two cents is that we should have a common Metadata class and it should be present on the schema of all attributes using Request.

Reply:
https://github.com/zytedata/zyte-common-items/pull/37/commits/a3f998f4961ede4aed8a5e3ee965d67ea4adc03c